src/js/tracks/text-track.js

```diff
@@ -420,6 +420,36 @@ class TextTrack extends Track {
     this.cues.setCues_(this.cues_);
   }
 
+  /**
+   * Makes the text track serializable by removing circular dependencies.
+   *
+   * @return {Object} The track information as a serializable object
+   */
+  toJSON() {
```

Comment:
we already have en existing implementation here: https://github.com/videojs/video.js/blob/8842d376ed3d834ad3ce0a5e662fe446513ea0bd/src/js/tracks/text-track-list-converter.js#L21
should it be used instead? Are there properties we'd want to keep that isn't in that list?

Reply:
Great question (and this was not a function I realized existed 😅). The newer implementation does contain more properties (activeCues, cues), but not sure how vital this information is for the serialization.

I do not have a strong preference on which functionality to use to determine the serialized data (I lean towards having the extra properties), but I do think if we decide to use the old method, we should move it to the `TextTrack` class so it can be public.

Reply:
I think the best bet is to have this call out to `trackToJson_`, since usage of text track list converter can end up working with the native text track object as well.

I don't think `activeCues` matters, but that could be added similarly to how trackToJson_ currently handles `cues`

Reply:
Makes sense, I updated to use the `trackToJson` method. I did update the name from `trackToJson_` as _I think_ that was trying to signal a private method, when it is not.